src/wam2layers/preprocessing/era5.py

```diff
@@ -185,45 +193,63 @@ def get_dp_pressurelevels(q, u, v, ps, qs, us, vs):
         v interpolated to new pressure levels,
         Pressure at the division between the two layers
     """
-    assert u.dims[0] == "level"  # move to better location
+    level_ax: int = u.get_axis_num("level")  # type: ignore
+    spatial_axes = u.get_axis_num(("latitude", "longitude"))
 
     p = u["level"].broadcast_like(u) * 100  # Pa
     p.attrs["units"] = "Pa"
 
+    _p = p.to_numpy()
+    _u = u.to_numpy()
+    _v = v.to_numpy()
+    _q = q.to_numpy()
```

Comment:
Does this create copies, i.e. does it double the memory use? If so, could temporarily delete xr arrays until you convert back on line 247

Reply:
With the changes to the code structure I am not sure if it's a good idea to delete the DataArrays anymore.

To reduce the memory use, we can load the DataArrays lazily (in the `load_data` function).